packages/amplify-graphql-relational-transformer/src/schema.ts

```diff
@@ -126,7 +126,10 @@ const ensureModelSortDirectionEnum = (ctx: TransformerContextProvider): void =>
 /**
  * ensureHasOneConnectionField
  */
-export const ensureHasOneConnectionField = (config: HasOneDirectiveConfiguration, ctx: TransformerContextProvider): void => {
+export const ensureHasOneConnectionField = (
+  config: HasOneDirectiveConfiguration | BelongsToDirectiveConfiguration,
```

Comment:
Why is `BelongsTo..` added here? Is it being used by belongsTo process?

Reply:
This is needed because we've added an `indexName` property to the `HasOneDirectiveConfiguration`, which results in the following error:

> Argument of type 'BelongsToDirectiveConfiguration' is not assignable to parameter of type 'HasOneDirectiveConfiguration'.
  Property 'indexName' is missing in type 'BelongsToDirectiveConfiguration' but required in type 'HasOneDirectiveConfiguration'.

in the existing [ensureBelongsToConnectionField](https://github.com/aws-amplify/amplify-category-api/blob/ed44cd66d504e6d29fe74968d24ae1ce2f665d0a/packages/amplify-graphql-relational-transformer/src/schema.ts#L262) method.